### PR TITLE
doc: update PFM spec

### PR DESCRIPTION
**Pull request by renaynay**
I mostly made this PR to track some misunderstandings I had with the spec for using PFM.

### CODE REVIEW

File: x/payment/spec/docs.md [hunk 1]
```diff
@@ -113,15 +113,15 @@ if err != nil {
 }
 
 // query for account information necessary to sign a valid tx
-err = keyringSigner.QueryAccount(ctx, grpcClientConn)
+err = keyringSigner.QueryAccountNumber(ctx, grpcClientConn)
```
Comment (renaynay) [inline x/payment/spec/docs.md:116]: Is this the method we should be using?

Comment (evan-forbes) [reply x/payment/spec/docs.md:116]: this brings up a really good point. This method appears to be vestigial, and should likely not be exported. Instead, it should be private and only called during the keyringSigner's BroadcastTx method.

Comment (evan-forbes) [reply x/payment/spec/docs.md:116]: #294

Comment (renaynay) [reply x/payment/spec/docs.md:116]: For the moment though, is this a necessary component to building and submitting a PFD?

**Comment by liamsi** [comment]
what is the status of this PR?

**Comment by evan-forbes** [comment]
I think we can close, as I believe we covered this issue synchronously but will defer to @renaynay

**Pull request closed without merging**